src/Executables/PrintMinimumGridSpacing/PrintMinimumGridSpacing.cpp

```diff
@@ -0,0 +1,154 @@
+      options.template get<OptionTags::DomainCreator<Dim, frame>>();
+  auto domain = domain_creator->create_domain();
+  double min_grid_spacing = std::numeric_limits<double>::max();
+  for (const auto& block : domain.blocks()) {
+    const auto initial_ref_levs =
+        domain_creator->initial_refinement_levels()[block.id()];
+    const std::vector<ElementId<Dim>> element_ids =
+        initial_element_ids(block.id(), initial_ref_levs);
+    for (const auto& element_id : element_ids) {
+      ElementMap<Dim, frame> map(element_id,
+                                 block.coordinate_map().get_clone());
+      Mesh<Dim> mesh(domain_creator->initial_extents()[block.id()],
+                     Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto);
+      min_grid_spacing = std::min(
+          min_grid_spacing,
+          minimum_grid_spacing(mesh.extents(), map(logical_coordinates(mesh))));
+    }
+  }
+  Parallel::printf("The minimum grid spacing is: %1.14e\n", min_grid_spacing);
+}
+
+std::string custom_help_msg() {
+  constexpr int max_label_size = 21;
+  const std::string help_text =
+      "Print the minimum grid spacing between inertial coordinates of the "
+      "Domain specified in the input file. The output can be used to "
+      "choose appropriate time steps.";
+
+  std::ostringstream ss;
+  ss << "\n==== Description of expected options:\n" << help_text
+     << "\n\nOptions:\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << "Dimension"
+     << "Value\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << ""
+     << "The dimensions of the desired domain (1, 2, or 3).\n\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << "DomainCreator"
+     << "DomainCreator<Dim, Frame::Inertial>\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << ""
+     << "The domain to create initially.\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << ""
+     << "Dim must be equal to Dimension's Value.\n\n";
+
+  return ss.str();
+}
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+int main(int argc, char* argv[]) {
+  namespace bpo = boost::program_options;
+
+  bpo::options_description command_line_options;
+  command_line_options.add_options()
+      ("help,h", "Describe program options")
+      ("input-file", bpo::value<std::string>(), "Input file name");
+
+  bpo::command_line_parser command_line_parser(argc, argv);
+  command_line_parser.options(command_line_options);
+  command_line_parser.positional({});
+
+  bpo::variables_map parsed_command_line_options;
+  bpo::store(command_line_parser.run(), parsed_command_line_options);
+  bpo::notify(parsed_command_line_options);
+
+  if (parsed_command_line_options.count("help") != 0) {
+    Parallel::printf("%s\n%s", command_line_options, custom_help_msg());
```

Comment:
I think for this you can do:
```cpp
    printf(
        "%s\n%s", std::string{MakeString{} << command_line_options}.c_str(),
        Options<tmpl::list<Dimension, OptionTags::DomainCreator<1, frame>>>{
            "Print the minimum grid spacing between inertial coordinates of\n"
            "the Domain specified in the input file. The output can be used\n"
            "to choose appropriate time step size.\n"
            "Note: The dimensionality of the domain depends on the\n"
            "dimensionality specified as the command line argument."}
            .help()
            .c_str());
```
Note that you'll need `Utilities/MakeString.hpp`

Reply:
@nilsdeppe A question regarding this - I originally wrote `custom_help_msg` because I was concerned that doing something like you suggested here would require that a dimension is specified for the `DomainCreator`, and so the help message printed will also specify whatever placeholder dimension that's given (1 in this example). I was concerned with the help message suggesting that the placeholder dimension is the dimension you must use. This is what the help output would look like for the `DomainCreator` option with 1 as the placeholder:

```
  DomainCreator          DomainCreator<1ul, Frame::Inertial>
                         The domain to create initially
```

What are your thoughts on this? In addition, in your suggestion, the custom `Dimension` option I created in the executable and added to the yaml files is still specified. However, in another comment you requested that `dimension` be a command line argument. Do you mean for it to be both an option and a command line argument? Lastly, if it's meant to be a command line argument, then won't we still run into the issue with the `execute` tests since executing requires an additional command line argument on top of `--input-file` (i.e. the conversation we had that led to wanting to make dimension yaml option)?

Reply:
Yea, though I think it's better to have a note in the help string pointing out that any dimensionality is allowed (between 1 and 3). That is what my `dimension` is referring too, though clearly it could be worded better.

Oh right, I forgot about the command line argument. I'm looking at `cmake/AddInputFileTests.cmake` and I think it would make sense to have tho possibility to specify additional command line arguments in the input file. Similar to what is currently done with the `checks`. I can do this, or you can take a stab at it too if you're interested in learning more CMake (feel free to say no! :) )

Reply:
I think finding a way to specify command line arguments in the input file would be cool! If it were another time, I would love to try to tackle this and learn more about CMake, but I have a few other things on my to-do list for my summer at CSUF, so I don't think I should tack on something new at the moment and will leave it to you! And if this is the case that things will be changed so that extra command line arguments can be specified by the input file, then in the mean time should I remove my custom `Dimension` option and wait for further updates before reconfiguring parsing `dimension` as a command line argument?

Reply:
@nilsdeppe Pinging you on this in case it got lost in the shuffle. When you get a chance, let me know how you would like me to move forward with this or sit on it for now considering that https://github.com/sxs-collaboration/spectre/pull/1628 is still in progress! Also, @wthrowe made a suggestion in a comment below regarding this dimension issue

Reply:
I like the suggestion Will made. That seems like a good way to go :)